src/core/videosource.cpp

```diff
@@ -245,11 +245,41 @@ FFMS_VideoSource::FFMS_VideoSource(const char *SourceFile, FFMS_Index &Index, in
+            if (FormatContext->streams[VideoTrack]->side_data[i].type == AV_PKT_DATA_STEREO3D) {
+                const AVStereo3D *StereoSideData = (const AVStereo3D *)FormatContext->streams[VideoTrack]->side_data[i].data;
                 VP.Stereo3DType = StereoSideData->type;
                 VP.Stereo3DFlags = StereoSideData->flags;
-                break;
+            } else if (FormatContext->streams[VideoTrack]->side_data[i].type == AV_PKT_DATA_MASTERING_DISPLAY_METADATA) {
+                const AVMasteringDisplayMetadata *MasteringDisplay = (const AVMasteringDisplayMetadata *)FormatContext->streams[VideoTrack]->side_data[i].data;
+                if (MasteringDisplay->has_primaries) {
+                    VP.HasMasteringDisplayPrimaries = MasteringDisplay->has_primaries;
+                    for (int i = 0; i < 3; i++) {
+                        VP.MasteringDisplayPrimariesX[i] = av_q2d(MasteringDisplay->display_primaries[i][0]);
+                        VP.MasteringDisplayPrimariesY[i] = av_q2d(MasteringDisplay->display_primaries[i][1]);
+                    }
+                    VP.MasteringDisplayWhitePointX = av_q2d(MasteringDisplay->white_point[0]);
+                    VP.MasteringDisplayWhitePointY = av_q2d(MasteringDisplay->white_point[1]);
+                }
+                if (MasteringDisplay->has_luminance) {
+                    VP.HasMasteringDisplayLuminance = MasteringDisplay->has_luminance;
+                    VP.MasteringDisplayMinLuminance = av_q2d(MasteringDisplay->min_luminance);
+                    VP.MasteringDisplayMaxLuminance = av_q2d(MasteringDisplay->max_luminance);
+                }
+
+                VP.HasMasteringDisplayPrimaries = !!VP.MasteringDisplayPrimariesX[0] && !!VP.MasteringDisplayPrimariesY[0] &&
+                                                  !!VP.MasteringDisplayPrimariesX[1] && !!VP.MasteringDisplayPrimariesY[1] &&
+                                                  !!VP.MasteringDisplayPrimariesX[2] && !!VP.MasteringDisplayPrimariesY[2] &&
+                                                  !!VP.MasteringDisplayWhitePointX   && !!VP.MasteringDisplayWhitePointY;
+                /* MasteringDisplayMinLuminance can be 0 */
+                VP.HasMasteringDisplayLuminance = !!VP.MasteringDisplayMaxLuminance;
+            } else if (FormatContext->streams[VideoTrack]->side_data[i].type == AV_PKT_DATA_CONTENT_LIGHT_LEVEL) {
+                const AVContentLightMetadata *ContentLightLevel = (const AVContentLightMetadata *)FormatContext->streams[VideoTrack]->side_data[i].data;
+
+                VP.ContentLightLevelMax = ContentLightLevel->MaxCLL;
+                VP.ContentLightLevelAverage = ContentLightLevel->MaxFALL;
+
+                /* Only check for either of them */
+                VP.HasContentLightLevel = !!VP.ContentLightLevelMax || !!VP.ContentLightLevelAverage;
```

Comment:
Not too familiar with HDR stuff, but is it ever useful to say which?

Reply:
in practice it never happens that only one variable is set and the other is not, but other tools like x265 use the same check

`    m_emitCLLSEI = p->maxCLL || p->maxFALL;`